src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

```diff
@@ -393,6 +402,39 @@ public void CopyTo(T[] array, int arrayIndex)
             Array.Copy(_items, 0, array, arrayIndex, _size);
         }
 
+        public void CopyTo(int sourceIndex, int count, Span<T> destination)
+        {
+            if (sourceIndex < 0)
```

Comment:
Given this extensive argument validation, this API is likely going to slower (for small number of elements) than `CollectionMarshal.AsSpan(list).CopyTo(span)` that you can do today already. It makes me wonder whether these APIs are really worth it - added comment at https://github.com/dotnet/corefx/issues/33006#issuecomment-565879150.

Reply:
Where can I find the CollectionMarshal?

Reply:
You would need an additional Slide() what makes more validation and CopyTo also make a validation.
We are probably making less validations here, but I don't know if I would add this API (even if it is faster and more convenient) if we already have CollectionMarshal.AsSpan.

Reply:
> Slice() what makes more validation

Note that the validation in the Span methods is fine-tuned using various tricks. For example, `Slice` avoids one of the checks via a trick that requires 6 line comment to explain. (I do not think we want to be duplicating these tricks here or in other places.)